lib/database/sthDatabase.js

```diff
@@ -81,14 +81,17 @@ function getJSONCSVOptions(attrName) {
  *  - {string} dbURI The database URI
  *  - {string} replicaSet The replica set name, if any
  *  - {string} database The name of the database to connect to
+ *  - {string} authSource The name of the database for the authorization
  *  - {Number} poolSize The size of the pool of connections to the database
  * @param {Function} callback A callback to inform about the result of the operation
  */
 function connect(params, callback) {
     connectionURL =
         // prettier-ignore
-        'mongodb://' + params.authentication + '@' + params.dbURI + '/' + params.database + 
-            (params.replicaSet ? '?replicaSet=' + params.replicaSet : '');
+        'mongodb://' + params.authentication + '@' + params.dbURI + '/' + params.database +
+            (params.replicaSet ? '?replicaSet=' + params.replicaSet : '') +
+            (params.authSource ? (params.replicaSet ? '&authSource=' +
+                params.authSource : '?authSource=' + params.authSource) : '');
```

Comment:
I understand this logic ensures that only if authSource has been defined a `authSource=xxxx` token is added to the MongoDB URI.

What about `authSource == ''` (the empty string)? Would be `authSource` omitted from URI in this case? Or a `authSource=` should be added (which I understand is a wrong URI and should be avoided)?

Reply:
As I understand the query (`params.authSource ?`) and Javascript, the string is checked for null, undefined or empty. (https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in)

Reply:
Ok. NTC